### PR TITLE
Log Level Test Fix + Removed Direct HTTPs mode tests + Authorization Token Type Invalid Fix

### PR DESCRIPTION
**Pull request by kushagraThapar**
* Log Level Test Fix 
* Removed Direct HTTPs mode tests

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosContainerChangeFeedTest.java [hunk 1]
```diff
@@ -70,7 +70,7 @@
     private final Multimap<String, ObjectNode> partitionKeyToDocuments = ArrayListMultimap.create();
     private final String preExistingDatabaseId = CosmosDatabaseForTest.generateId();
 
-    @Factory(dataProvider = "simpleClientBuildersWithDirect")
+    @Factory(dataProvider = "simpleClientBuildersWithDirectTcp")
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosContainerChangeFeedTest.java:73]: You can just change the implementation of `simpleClientBuildersWithDirect` to remove DirectHttps that will be easier to do as you don't have to touch every test file to change `simpleClientBuildersWithDirect` to `simpleClientBuildersWithDirectTcp`.

the change will become mostly isolated only in one or two place

**Review by moderakh — COMMENTED** [review]
You can just change the implementation of `simpleClientBuildersWithDirect` to remove DirectHttps that will be easier to do as you don't have to touch every test file to change `simpleClientBuildersWithDirect` to `simpleClientBuildersWithDirectTcp`.

the change will become mostly isolated only in one or two place

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/TestSuiteBase.java [hunk 2]
```diff
@@ -90,7 +90,7 @@
                                           allEqualOrLowerConsistencies(accountConsistency)));
         preferredLocations = immutableListOrNull(parsePreferredLocation(TestConfigurations.PREFERRED_LOCATIONS));
         protocols = ObjectUtils.defaultIfNull(immutableListOrNull(parseProtocols(TestConfigurations.PROTOCOLS)),
-                                              ImmutableList.of(Protocol.HTTPS, Protocol.TCP));
+                                              ImmutableList.of(Protocol.TCP));
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/TestSuiteBase.java:93]: this is not the only place that we pass Direct Https.

Please see this:
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/TestConfigurations.java#L81-L85

you may need to change the test yaml files which populate this environment variable.

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/TestSuiteBase.java:93]: Yes, I will update the CI yml files as well. Thanks for catching this.

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosContainerChangeFeedTest.java (hunk 1 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosContainerChangeFeedTest.java:73]: Makes sense, will do that.

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosContainerChangeFeedTest.java]: Updated.

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/TestSuiteBase.java (hunk 2 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/TestSuiteBase.java:93]: Updated.

**Review by milismsft — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java [hunk 3]
```diff
@@ -379,4 +379,25 @@ private void releaseOnNotSubscribedResponse(ReactorNettyResponseState reactorNet
...
+    /**
+     * DO NOT USE
+     *
+     * This API is only for testing purposes, don't use it anywhere else in the code.
+     * This changes the logging level of Reactor Netty Http Client.
+     */
+    public void enableNetworkLogging() {
+        Logger logger = LoggerFactory.getLogger(REACTOR_NETWORK_LOG_CATEGORY);
+        if (logger.isTraceEnabled()) {
+            this.httpClient = this.httpClient.wiretap(REACTOR_NETWORK_LOG_CATEGORY, LogLevel.TRACE);
```
Comment (simplynaveen20) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java:392]: Can't we do it in the test, we can use reflection to get httpClient  out of ReactorNettyClient and then do the same thing there ?

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java:392]: We can, but wanted to do it cleanly in one place. Also, this API will be used for further improvement for Reactor Netty Client logging. Current implementation has some restrictions on what logging customer can use for this Http Client. So added this API which will be used in future for updating the Reactor netty http client logging options.

Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java:392]: I would introduce it in main code when needed for customer use. Right now we can create a method in some base test class and use every where else

**Review by moderakh — APPROVED** [review]
LGTM. thanks Kushagra

**Review by xinlian12 — APPROVED** [review]
LGTM. Thanks a lot for the fix

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java (hunk 3 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java:392]: In my opinion, using some reflection to update the config options of http client is not clean way. We will need to use reflection to get internal `httpClient` from `ReactorNettyClient` which just complicates this simple test. 

Also, `ReactorNettyClient` is already an implementation class, so creating this API is not going to affect public surface area in any way.

Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java:392]: No strong opinion , so please go ahead with current. All set

**Review by simplynaveen20 — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java (hunk 3 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java:392]: Thanks @simplynaveen20

**Pull request merged**